src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs

```diff
@@ -416,7 +417,7 @@ private void CheckUnarySignature(DiagnosticBag diagnostics)
             // SPEC: A unary + - ! ~ operator must take a single parameter of type
             // SPEC: T or T? and can return any type.
 
-            if (this.ParameterTypes[0].StrippedType() != this.ContainingType)
+            if (this.ParameterTypes[0].StrippedType().TupleUnderlyingTypeOrSelf() != this.ContainingType)
```

Comment:
It is not up to us to decide. One should be able to declare whatever operators inside ValueTuple type definition.

---
In reply to: [63385518](https://github.com/dotnet/roslyn/pull/11328#discussion_r63385518) [](ancestors = 63385518)